packages/react-components/components/SettingsItem.tsx

```diff
@@ -1,52 +1,179 @@
-import Touchable from '@celo/react-components/components/Touchable'
+import ListItem from '@celo/react-components/components/ListItem'
+import TextInput from '@celo/react-components/components/TextInput.v2'
 import ForwardChevron from '@celo/react-components/icons/ForwardChevron'
-import colors from '@celo/react-components/styles/colors'
-import fontStyles from '@celo/react-components/styles/fonts'
+import colors from '@celo/react-components/styles/colors.v2'
+import fontStyles from '@celo/react-components/styles/fonts.v2'
 import * as React from 'react'
-import { StyleSheet, Text, View } from 'react-native'
+import { StyleSheet, Switch, Text, View } from 'react-native'
 
-export interface SettingsItemProps {
+interface WrapperProps {
   testID?: string
-  title?: string
-  children?: React.ReactChild
   onPress?: () => void
+  children: React.ReactNode
 }
 
-class SettingsItem extends React.Component<SettingsItemProps> {
-  render() {
-    const { testID, title, onPress, children } = this.props
+function Wrapper({ testID, onPress, children }: WrapperProps) {
+  return (
+    <ListItem testID={testID} onPress={onPress}>
+      {children}
+    </ListItem>
+  )
+}
 
-    return (
-      <Touchable testID={testID} onPress={onPress}>
-        <View style={style.settingsItemContainer}>
-          <View style={[style.title]}>
-            {children ? children : <Text style={fontStyles.body}>{title}</Text>}
-          </View>
-          <View style={[style.cta]}>
-            <ForwardChevron height={16} />
-          </View>
+function Title({ value }: { value: string }) {
+  return <View style={[styles.left]}>{<Text style={styles.title}>{value}</Text>}</View>
+}
+
+type SettingsItemNoValueProps = {
+  title: string
+} & Omit<WrapperProps, 'children'>
+
+export function SettingsItemNoValue({ testID, title, onPress }: SettingsItemNoValueProps) {
```

Comment:
i'm all for being explicit, but the `NoValue` in the name seems to imply that value could be an optional prop? i'm okay with this if it's truly ugly, but as a consumer it feels easier to just import one component as opposed to knowing which one i have to use

Reply:
maybe SettingsItem has an enum type?

Reply:
Good idea, I will merge `SettingsItemNoValue` and `SettingsItemTextValue` into one component, but would keep `SettingsItemSwitch` and `SettingsItemInput` separate. WDYT?